### PR TITLE
 3.8  version OHOS adaptation

### PR DESCRIPTION
**Pull request by xqg243795168**
3.8  version OHOS adaptation

### CODE REVIEW

File: cocos/renderer/CCGLProgramState.cpp [hunk 1]
```diff
@@ -363,8 +363,11 @@ GLProgramState::GLProgramState()
 , _vertexAttribsFlags(0)
 , _glprogram(nullptr)
 , _nodeBinding(nullptr)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+, _backToForegroundlistener(nullptr)
+#endif
```
Comment (minggo) [inline cocos/renderer/CCGLProgramState.cpp]: Why only initialize on OHOS?

File: cocos/renderer/CCMeshCommand.cpp [hunk 2]
```diff
@@ -56,10 +56,13 @@ MeshCommand::MeshCommand()
 , _vao(0)
 , _material(nullptr)
 , _stateBlock(nullptr)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+, _rendererRecreatedListener(nullptr)
+#endif
```
Comment (minggo) [inline cocos/renderer/CCMeshCommand.cpp]: Why only initialize on OHOS?

File: tests/lua-tests/project/Classes/AppDelegate.cpp [hunk 3]
```diff
@@ -43,16 +43,16 @@ bool AppDelegate::applicationDidFinishLaunching()
...
         register_assetsmanager_test_sample(L);
 #endif
         register_test_binding(L);
     }
     lua_pop(L, 1);
 
 
-    pEngine->executeScriptFile("src/controller.lua");
-
+    FileUtils::getInstance()->addSearchPath("");
```
Comment (minggo) [inline tests/lua-tests/project/Classes/AppDelegate.cpp]: Why add an empty path?

File: cocos/renderer/CCGLProgramState.cpp (hunk 1 above)
Comment (wanghui187) [reply cocos/renderer/CCGLProgramState.cpp]: fixed

File: cocos/renderer/CCMeshCommand.cpp (hunk 2 above)
Comment (wanghui187) [reply cocos/renderer/CCMeshCommand.cpp]: fixed

File: tests/lua-tests/project/Classes/AppDelegate.cpp (hunk 3 above)
Comment (wanghui187) [reply tests/lua-tests/project/Classes/AppDelegate.cpp]: fixed

File: tests/lua-tests/project/Classes/AppDelegate.cpp [hunk 4]
```diff
@@ -51,8 +51,10 @@ bool AppDelegate::applicationDidFinishLaunching()
     lua_pop(L, 1);
 
 
-    FileUtils::getInstance()->addSearchPath("");
-    pEngine->executeScriptFile("controller.lua");
+	#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+    	pEngine->executeScriptFile("controller.lua");
+	#else
+	    pEngine->executeScriptFile("src/controller.lua");
```
Comment (minggo) [inline tests/lua-tests/project/Classes/AppDelegate.cpp]: Wrong indent.

**Pull request merged**